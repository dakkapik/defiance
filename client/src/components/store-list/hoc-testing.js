import React from "react";
import { render } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import storesReducer from "../../redux/stores/stores.reducer";

export const renderWithState = (
  ui,
  { initialState, ...renderOptions } = {}
) => {
  const store = createStore(storesReducer, initialState);
  const Wrapper = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );

  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export default renderWithState;
