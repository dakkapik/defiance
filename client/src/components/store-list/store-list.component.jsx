import React, { useEffect } from "react";
import StoreItem from "../store-item/store-item.component";
import { connect } from "react-redux";
import { fetchApiStoreListUseEffect } from "../../redux/stores/stores.action";
import "./store-list.styles.scss";
export const StoreList = ({ stores, fetchApiStoreListUseEffect }) => {
  useEffect(() => {
    fetchApiStoreListUseEffect();
  }, [fetchApiStoreListUseEffect]);
  return (
    <div className="store-list">
      {stores.map((storeinfo, index) => (
        <StoreItem key={storeinfo._id} storeInfo={storeinfo} />
      ))}
    </div>
  );
};

export const mapStateToProps = (state) => ({
  stores: state.stores.stores,
});

export const mapDispatchToProps = (dispatch) => ({
  fetchApiStoreListUseEffect: () => dispatch(fetchApiStoreListUseEffect()),
});
export default connect(mapStateToProps, mapDispatchToProps)(StoreList);
