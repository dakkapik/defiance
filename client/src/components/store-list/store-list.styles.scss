.store-list {
  display: flex;
  float: left;
  flex-wrap: wrap;
  flex-direction: column;

  justify-content: space-evenly;

  border-radius: 10px;
  width: 20%;
  color: white;
  @media screen and (max-width: 900px) {
    width: unset;
    float: unset;
    padding: 5vh;
    display: flex;
    flex-direction: unset;
  }
}
