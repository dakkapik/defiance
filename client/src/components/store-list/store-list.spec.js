import React from "react";
import { StoreList } from "./store-list.component";
import { shallow } from "enzyme";

describe("StoreList component render", () => {
  let wrapper;

  let mocktriggerStores;

  beforeEach(() => {
    mocktriggerStores = jest.fn();

    const mockProps = {
      stores: [
        {
          location: { lat: 26.260501899508107, lng: -80.26398159301463 },

          manager: false,
          name: "Royal Palm",
          storeId: "psq1",
          __v: 0,
          _id: "60410e74be5b952498d3a376",
        },
      ],
      triggerStores: mocktriggerStores,
    };
    wrapper = shallow(<StoreList {...mockProps} />);
  });

  it("should render StoreList component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
