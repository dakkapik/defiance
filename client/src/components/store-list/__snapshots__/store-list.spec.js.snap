// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreList component render should render StoreList component 1`] = `ShallowWrapper {}`;
